pkg/receive/multitsdb.go

```diff
@@ -84,6 +86,7 @@ func NewMultiTSDB(
 	bucket objstore.Bucket,
 	allowOutOfOrderUpload bool,
 	hashFunc metadata.HashFunc,
+	metricNameFilterEnabled bool,
```

Comment:
It could be a good idea to apply the same pattern we have for proxy store: https://github.com/thanos-io/thanos/blob/main/pkg/store/proxy.go#L141. This way adding new options will not require changing so many places in the code base.